optuna/integration/sklearn.py

```diff
@@ -513,11 +513,12 @@ def _estimator_type(self) -> str:
 
     @property
     def best_index_(self) -> int:
-        """Index which corresponds to the best candidate parameter setting."""
+        """Trial number which corresponds to the best candidate parameter setting.
 
-        df = self.trials_dataframe()
+        Retuned value is equivant to ``optuna_search.best_trial_.number``.
+        """
 
-        return df["value"].idxmin()
+        return self.best_trial_.number
```

Comment:
Inside of `best_trial_` calls `self._check_is_fitted()`, so we do not need to call `self._check_is_fitted()` in  `best_index_`.